OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -771,6 +771,43 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				if (engineVersion < 20170628)
+				{
+					var building = node.Value.Nodes.FirstOrDefault(n => n.Key == "Building");
+					var bib = node.Value.Nodes.FirstOrDefault(n => n.Key == "Bib");
+
+					var hasBib = false;
+					if (bib != null)
+					{
+						var minibib = bib.Value.Nodes.FirstOrDefault(n => n.Key == "HasMinibib");
+						if (minibib != null)
+							hasBib = !FieldLoader.GetValue<bool>("HasMinibib", minibib.Value.Value);
+						else
+							hasBib = true;
+
+						Console.WriteLine("Bibs are no longer automatically included in building footprints. Please check if any manual adjustments are needed.");
+						RenameNodeKey(bib, "WithBuildingBib");
+					}
+
+					if (building != null && hasBib)
+					{
+						var footprint = building.Value.Nodes.FirstOrDefault(n => n.Key == "Footprint");
+						var dimensions = building.Value.Nodes.FirstOrDefault(n => n.Key == "Dimensions");
+						if (footprint != null && dimensions != null)
+						{
+							var newDim = FieldLoader.GetValue<CVec>("Dimensions", dimensions.Value.Value) + new CVec(0, 1);
+							var oldFootprint = FieldLoader.GetValue<string>("Footprint", footprint.Value.Value);
+							var addRow = new string[] { " =", " ==", " ===", " ====" };
+							dimensions.Value.Value = newDim.ToString();
+							footprint.Value.Value = oldFootprint + addRow[newDim.X - 1];
```

Comment:
This will break for mods that use >4 cell width actors.  You should be able to use something like `+ " " + string.Concat(Enumerable.Repeat("=", newDim.X))`